rs/ethereum/ledger-suite-orchestrator/src/state/mod.rs

```diff
@@ -445,16 +535,21 @@ impl State {
         &mut self.cycles_management
     }
 
-    pub fn managed_canisters_iter(&self) -> impl Iterator<Item = (&Erc20Token, &Canisters)> {
-        self.managed_canisters.canisters.iter()
+    pub fn all_managed_canisters_iter(&self) -> impl Iterator<Item = (TokenId, &Canisters)> {
+        self.managed_canisters.all_canisters_iter()
+    }
+
+    pub fn all_managed_principals(&self) -> impl Iterator<Item = &Principal> {
+        self.all_managed_canisters_iter()
+            .flat_map(|(_, canisters)| canisters.principals_iter())
     }
 
-    pub fn managed_erc20_tokens_iter(&self) -> impl Iterator<Item = &Erc20Token> {
-        self.managed_canisters.canisters.keys()
+    pub fn all_managed_tokens_ids_iter(&self) -> impl Iterator<Item = TokenId> + '_ {
```

Comment:
What's `+ '_`?

Reply:
That's a bound on the lifetime that's elided, see https://doc.rust-lang.org/reference/lifetime-elision.html. So that's the same as
```rust
 pub fn all_managed_tokens_ids_iter<'a>(&'a self) -> impl Iterator<Item = TokenId> + 'a {
```